packages/client/src/comp/states/Base.tsx

```diff
@@ -23,6 +26,34 @@ type Props<Info> = React.PropsWithChildren<{
   enrichValues: EnrichValues<Info>
 }>
 
+const datePattern = /^([0-3]\d{1})\/((0|1|2)\d{1})\/((19|20)\d{2})/
```

Comment:
American dates are month-day-year

Also, no month starts with 2.  Let's make dates be less than 32 and only 1-12 be the only valid months.